src/rgw/rgw_sal_motr.cc

```diff
@@ -2624,7 +2624,7 @@ int MotrMultipartUpload::delete_parts(const DoutPrefixProvider *dpp)
   // Delete object part index.
   std::string oid = mp_obj.get_key();
   string tenant_bkt_name = get_bucket_name(bucket->get_tenant(), bucket->get_name());
-  string obj_part_iname = "motr.rgw.object." + tenant_bkt_name + "." + oid + ".parts";
+  string obj_part_iname = "motr.rgw.object." + tenant_bkt_name + "." + oid + "." + get_upload_id() + ".parts";
```

Comment:
Do we have upload id available once the multipart upload is complete ? Will this change in bucket name will work across all states in the life multipart object ?

Reply:
once multipart upload is completed this part table is no longer accessible. And the upload id is available till multipart lifecycle i.e till abort/complete.

Reply:
@sachinpunadikar i have added a testing dropdown to showcase my testing results